src/target/riscv/riscv-011.c

```diff
@@ -1337,6 +1337,10 @@ static int get_register(struct target *target, riscv_reg_t *value,
 	maybe_write_tselect(target);
 
 	if (regid <= GDB_REGNO_XPR31) {
+		/* FIXME: Here the implementation assumes that the value
+		 * written to GPR will be the same as the value read back. This
+		 * is not true for a write of a non-zero value to x0.
+		 */
 		*value = reg_cache_get(target, regid);
```

Comment:
If the below "if" in a below comment is added, it looks to me the issue should be resolved and this comment can be simply dropped. 

- The reads of GPR x0 would return the value obtained on the last halt (that is, always `0x0` on correct hardware).
- The reads of other GPRs would return what is in the cache -- from last halt or from last write.

Again, please correct me if I am missing something.

```suggestion
	if (regid <= GDB_REGNO_XPR31) {
		*value = reg_cache_get(target, regid);
```

Reply:
I think you are right. Filed an issue: #1086

Reply:
Ok, thanks.